src/3d/chunks/qgschunkedentity_p.cpp

```diff
@@ -399,7 +402,8 @@ void QgsChunkedEntity::startJobs()
     delete entry;
 
     QgsChunkQueueJob *job = startJob( node );
-    mActiveJobs.append( job );
+    if ( job )
+      mActiveJobs.append( job );
```

Comment:
Is this addition necessary?

There never should be null jobs coming from startJob()... We should make sure elsewhere that we do not create jobs for any entities that are not valid...

Reply:
> Is this addition necessary?

I am not very sure they are necessary, if the but I was thinking that those guards could not hurt, and help easily avoid crash. I like guards for that!
A case where a job can be null it is when the mesh layer is null here : https://github.com/qgis/QGIS/blob/9aafc08bbabdd670adaf8e7704021987c677dbc2/src/3d/mesh/qgsmeshterraingenerator.cpp#L58
That could happen if the mesh layer in the mesh terrain generator is not valid anymore (removed by user for example). But, as the unique (for now) tile terrain entity is created directly with mesh layer information in the main thread (even if implicitly shared triangular mesh is used in other thread), there should not be problem...
So I remove those extra guards and cross finger that it always will be ok!

Reply:
Thanks... In the mesh terrain generator we could have `assert( mLayer )` so that we never return null pointer and make it clear that the function should not return null pointer - and that there should be code that ensures the function is not called with invalid terrain generator...